Source/WTF/wtf/JSONValues.h

```diff
@@ -156,6 +157,11 @@ class WTF_EXPORT_PRIVATE Value : public RefCounted<Value> {
     }
 
 private:
+    template<typename Visitor> constexpr decltype(auto) visitDerived(Visitor&&);
+    template<typename Visitor> constexpr decltype(auto) visitDerived(Visitor&&) const;
```

Comment:
NIT: do these actually need to be part of the class?  i.e. could they instead be `static` functions inside the `.cpp` that accept a `Value&` parameter?

Reply:
I personally like this (having it in JSON::Value itself) since how to dispatch this is largely depending on how we construct the class hierarchy of Value.